Islandora/islandora
# issue #959: Cache TIFF width and height into fields on File media if  they exist

#959 



# What does this Pull Request do?

When generating IIIF manifests in the Views Style plugin, if it has to resort to querying Cantaloupe, it will cache the result in fields on the file media if they exist.

# What's new?

Querying Cantaloupe for an image's width and height turns out to be costly performance-wise, and if it's done for multiple images such as when we generate a IIIF manifest out of Original Files, it can take several seconds to a minute or even time out if there are more than about 100 pages.

This change will store the height and width it finds own fields named 'field_height' and 'field_width' on the File media entity if those fields exist.

Accompanying this PR will be a starter site PR to add those fields by default.


* Does this change add any new dependencies?  No
* Does this change require any other modifications to be made to the repository 
 (i.e. Regeneration activity, etc.)?  No
* Could this change impact execution of existing code? generating manifests will be affected, but end result will be the same.

# How should this be tested?

An exported starter site config will be attached to this PR but is not necessary if you are comfortable making the following change:

1. Go to admin/structure/media and click 'Manage Fields' for the File media entity type.
2. Choose Add Existing Field and add the existing field_width and field_height fields that belong to the Image media entity type.
3. Save the changes

* if those fields are not present, this code just skips the caching process.

Then to test:.

1. Go to /node/add and create an object of type Repository Item with model Paged Content.
4. Add at least one Repository Item object as a member of that object, with model Page.
5. Add media to the child objects and upload a TIFF file for each, 
6. If you like, monitor the CPU usage on the Cantaloupe server.
7. Go to /node/xxx/book-manifest-original (if using the standard starter site where this is available) where did is the node of the Paged Content object you created in step 1.
8. Hit 'Refresh' after you see the manifest, it should generate much more quickly. For more robust testing, clear the site cache and then do the same.
9. Navigate to a media from this collection of objects and look for the Width and Height fields, they should now be populated.

# Documentation Status

* Does this change existing behaviour that's currently documented? No
* Does this change require new pages or sections of documentation? No
* Who does this need to be documented for?
* Associated documentation pull request(s): ___  or documentation issue ___

# Additional Notes:
Any additional information that you think would be helpful when reviewing this
 PR.

# Interested parties
Tag (@ mention) interested parties or, if unsure, @Islandora/committers
